### PR TITLE
[4] webservices vs custom field list

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #38642 .

### Summary of Changes
fixed

### Testing Instructions
see #38642

### Actual result BEFORE applying this Pull Request

500 

### Expected result AFTER applying this Pull Request
no error  
and return the selected field list values

### CODE REVIEW

**Comment by hokislotdemo** [comment]
Ini adalah pertanyaan paling favorit alias setiap hari muncul. Untuk kemenangan dari bermain di akun [link slot demo lengkap](https://joannabriggs.org/bola/slot-demo/ "") tidak bisa di tarik karena bersifat demo. Anda cukup mencoba bermain saja namun semua kemenangan tidak bisa di ambil. Dan untuk yang bisa di tarik tentu para pemain sudah harus melakukan deposit. Akan kami jelaskan di pertanyaan selanjutnya tentang game slot uang asli. Yang pasti akun demo jika menang semua tidak bisa di uangkan.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38655">issues.joomla.org/tracker/joomla-cms/38655</a>.</sub>

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 3bea502e631c5a5ebc161ee649e44e3ae5e441e3<br /><br />Tested successfully in Joomla 4.2.3-dev of 5 September in Wampserver 3.2.9 and PHP 8.1.9.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38655">issues.joomla.org/tracker/joomla-cms/38655</a>.</sub>

File: plugins/fields/radio/radio.php [hunk 1]
```diff
@@ -41,9 +41,21 @@ public function onCustomFieldsBeforePrepareField($context, $item, $field)
         if (!$this->isTypeSupported($field->type)) {
             return;
         }
+        
+        $field->apivalue = [];
```
Comment (richard67) [inline plugins/fields/radio/radio.php]:
```suggestion

        $field->apivalue = [];
```
Remove white space from empty line to fix code style error reported by drone.

File: plugins/fields/list/list.php [hunk 2]
```diff
@@ -44,7 +44,17 @@ public function onCustomFieldsBeforePrepareField($context, $item, $field)
 
         $options = $this->getOptionsFromField($field);
 
-        $field->apivalue = [$field->value => $options[$field->value]];
+       if (empty($field->value)) {
```
Comment (richard67) [inline plugins/fields/list/list.php]:
```suggestion
        if (empty($field->value)) {
```
Fix code style errors reported by drone.

**Review by antonbag — APPROVED** [review]

**Comment by Quy** [comment]
I don't know if it matters that the return format is different when no values.

Before PR (returns JSON):
```
            "myfields": {
                "": null
            },
            "radio": {
                "": null
            }
```

After PR (returns array):
```
            "myfields": [
                ""
            ],
            "radio": []
```

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 3f2c307e581d25faa8aeec35b05db44ddc2f4083<br /><br />Tested successfully in Joomla 4.2.9-rc2-dev of 13 March using PHP 8.1.10 and Postman v10.11.2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38655">issues.joomla.org/tracker/joomla-cms/38655</a>.</sub>

**Comment by alikon** [comment]
conflict solved on #41777

**Pull request closed without merging**